File: articles/institutional-responses-to-public-goods-problems/go/public_goods_score.go
package main

import "fmt"

type PublicGoodsCase struct {
	ContributionRate        float64
	Legitimacy              float64
	Trust                   float64
	Coordination            float64
	Monitoring              float64
	DistributionalAttention float64
	ScaleComplexity         float64
	CaptureRisk             float64
}

func PublicGoodsProvisionScoreRaw(x PublicGoodsCase) float64 {
	return 0.22*x.ContributionRate +
		0.13*x.Legitimacy +
		0.12*x.Trust +
		0.11*x.Coordination +
		0.10*x.Monitoring +
		0.08*x.DistributionalAttention -
		0.12*x.ScaleComplexity -
		0.08*x.CaptureRisk
}

func main() {
	demo := PublicGoodsCase{70, 78, 75, 72, 80, 74, 35, 25}
	fmt.Printf("Public goods provision raw score: %.2f\n", PublicGoodsProvisionScoreRaw(demo))
}

File: articles/institutional-responses-to-public-goods-problems/rust/public_goods_score.rs
struct PublicGoodsCase {
    contribution_rate: f64,
    legitimacy: f64,
    trust: f64,
    coordination: f64,
    monitoring: f64,
    distributional_attention: f64,
    scale_complexity: f64,
    capture_risk: f64,
}

fn public_goods_provision_score_raw(x: &PublicGoodsCase) -> f64 {
    0.22 * x.contribution_rate
        + 0.13 * x.legitimacy
        + 0.12 * x.trust
        + 0.11 * x.coordination
        + 0.10 * x.monitoring
        + 0.08 * x.distributional_attention
        - 0.12 * x.scale_complexity
        - 0.08 * x.capture_risk
}

fn main() {
    let demo = PublicGoodsCase {
        contribution_rate: 70.0,
        legitimacy: 78.0,
        trust: 75.0,
        coordination: 72.0,
        monitoring: 80.0,
        distributional_attention: 74.0,
        scale_complexity: 35.0,
        capture_risk: 25.0,
    };

    println!(
        "Public goods provision raw score: {:.2}",
        public_goods_provision_score_raw(&demo)
    );
}
